#2219 Complete setting definition of docs (#2331)
oharastream/ohara: docs/src/custom_streamapp.rst

which are described below for your convenience.

The following example is a simple streamApp application which can run in
Ohara. Note that this example simply starts the streamApp application without doing any transformation but writing data,
i.e., the target topic will have same data as the source topic.

.. code-block:: java

   public class SimpleApplicationForOharaEnv extends StreamApp {

     @Override
     public void start(OStream<Row> ostream, StreamDefinitions streamDefinitions) {
       ostream.start();
     }
   }

.. note::

>    The following methods we provided belong to Ohara StreamApp, which have

Reviewer: ```suggestion
   The following methods we provided belongs to Ohara StreamApp, which has
```

user to initialize some external data source connections or input
parameters.

.. _streamapp-config-method:

config() method
~~~~~~~~~~~~~~~

> In a streamApp application, you may want to config your own parameters. We support a method here to help you define

Reviewer: ```suggestion
In a streamApp application, you may want to configure your own parameters. We support a method here to help you define
```

> a custom streamDefinitions list in streamApp. The details of streamDefinitions is list :ref:`here <streamapp-setting-definitions>`.

Reviewer: ```suggestion
a custom streamDefinitions list in streamApp. The details of streamDefinitions are list :ref:`here <streamapp-setting-definitions>`.
```

.. _streamapp-setting-definitions:

StreamApp Definitions
---------------------

StreamApp stores a list of :ref:`SettingDef <setting-definition>`, which is StreamDefinitions, in the data store.

> By default, we will keep the following definitions in "core" group and generate the definition in stream api :

Reviewer: ```suggestion
By default, we will keep the following definitions in the "core" group and generate the definition in stream API :
```

#. DefaultConfigs.BROKER_DEFINITION : The broker list
#. DefaultConfigs.IMAGE_NAME_DEFINITION : The image name
#. DefaultConfigs.NAME_DEFINITION : The streamApp application name
#. DefaultConfigs.GROUP_DEFINITION : The streamApp group name
#. DefaultConfigs.FROM_TOPICS_DEFINITION : The from topic
#. DefaultConfigs.TO_TOPICS_DEFINITION : The to topic

> #. DefaultConfigs.JMX_PORT_DEFINITION : The expose jmx port

Reviewer: ```suggestion
#. DefaultConfigs.JMX_PORT_DEFINITION : The exposed jmx port
```

#. DefaultConfigs.INSTANCES_DEFINITION : The running instances
#. DefaultConfigs.NODE_NAMES_DEFINITION : The node name list
#. DefaultConfigs.VERSION_DEFINITION : The version of streamApp
#. DefaultConfigs.REVISION_DEFINITION : The revision of streamApp
#. DefaultConfigs.AUTHOR_DEFINITION : The author of streamApp
#. DefaultConfigs.TAGS_DEFINITION : The tags of streamApp

> Any other definition except above list will be trick as custom definition. You can define

Reviewer: ```suggestion
Any other definition except above list will be treated as a custom definition. You can define
```

.. code-block:: java

   SettingDef.builder().key(joinTopic).group("default").build()

> as a definition that is list in "default" group, or

Reviewer: ```suggestion
as a definition that is listed in "default" group, or
```

.. code-block:: java

   SettingDef.builder().key(otherKey).group("common").build()

> as a definition that is list in "common" group.

Reviewer: ```suggestion
as a definition that is listed in the "common" group.
```
